now/dialog.py

```
@@ -38,6 +38,25 @@ def configure_option(
 ) -> DialogStatus:
     # Check if it is dynamic. If it is then spawn multiple dialogs for each option and return with continue
     if option.dynamic_func:
+        if option.name in kwargs:
+            # Expand dynamic options from parent option, expect a dict (supports only model_selection)
+            for user_selection in kwargs[option.name].split(","):
+                if ":" in user_selection:
+                    option_name, option_values = user_selection.split(":")
+                    kwargs[f"{option_name}_model"] = []
+                    for option_value in option_values.split("+"):
+                        kwargs[f"{option_name}_model"].append(
```

We can wrap this up in a try-catch block or perform an `if` check on `options_value` and raise a proper message in case the user, provided some wrong model name.

---

RIght so initially I wanted to catch errors and raise something to the user. For this I wanted to leverage `parser` from `argparse`, but since we build it in other places I would have to pass it along all the functions all the way to code where we configure the dynamic options here... It was a lot of overhead. Thus I thought at first, we could silently ignore mistyped options. But I agree this is not ideal. What are your thoughts about this?

---

If it is possible, then we could pop the value from kwargs and raise `RetryException` (with a message that the passed fields are not correct, please select) so that this dialog will be asked to the user

---

I won't throw a `RetryException` here as this one is used only in the interactive context, while the parsing I added occurs when passing a selection a a command line argument. However, I will display nicer error in case of an error with the selection.

---

Ok I handle now two type of errors:

```
python main.py start --dataset_type path --dataset_path ~/Downloads/gif_err --index_fields .gif --model_selection foo.gif:Clip
```
```
ValueError: Error with --model_selection: foo.gif is not in an index.
```

and

```
python main.py start --dataset_type path --dataset_path ~/Downloads/gif_err --index_fields .gif --model_selection .gif:Sbert
```
```
ValueError: Error with --model_selection: Sbert is not available. for index .gif.
```

---

The error message looks much better now. Can we also print possible options that they could select in the same error message?
`ValueError: Error with --model_selection: Sbert is not available. for index .gif. Choose one/many among {..possible models..}`